internal/consul/registration.go

```diff
@@ -50,22 +53,72 @@ func NewServiceRegistry(logger hclog.Logger, consul *api.Client, service, namesp
 		tries:                  defaultMaxAttempts,
 		backoffInterval:        defaultBackoffInterval,
 		reregistrationInterval: 30 * time.Second,
+		updateTTLInterval:      10 * time.Second,
 	}
 }
 
+// WithTags adds tags to associate with the service being registered.
+func (s *ServiceRegistry) WithTags(tags []string) *ServiceRegistry {
+	s.tags = tags
+	return s
+}
+
 // WithTries tells the service registry to retry on any remote operations.
 func (s *ServiceRegistry) WithTries(tries uint64) *ServiceRegistry {
 	s.tries = tries
 	return s
 }
 
+// Register registers a Gateway service with Consul.
+func (s *ServiceRegistry) RegisterGateway(ctx context.Context) error {
+	return s.register(ctx, &api.AgentServiceRegistration{
+		Kind:      api.ServiceKind(api.IngressGateway),
+		ID:        s.id,
+		Name:      s.name,
+		Namespace: s.namespace,
+		Address:   s.host,
+		Tags:      s.tags,
+		Meta: map[string]string{
+			"external-source": "consul-api-gateway",
+		},
+		Checks: api.AgentServiceChecks{{
+			Name:                           fmt.Sprintf("%s - Ready", serviceCheckName),
+			TCP:                            fmt.Sprintf("%s:%d", s.host, 20000),
+			Interval:                       serviceCheckInterval,
+			DeregisterCriticalServiceAfter: serviceDeregistrationTime,
+		}},
+	}, false)
+}
+
 // Register registers a service with Consul.
 func (s *ServiceRegistry) Register(ctx context.Context) error {
```

Comment:
Is this currently being used only for registering the API Gateway controller service?